daemon/network.go

```diff
@@ -334,6 +335,11 @@ func (daemon *Daemon) GetNetworkDriverList() map[string]bool {
 	// TODO : Replace this with proper libnetwork API
 	pluginList["overlay"] = true
 
+	remotes, _ := plugins.GetAll("NetworkDriver")
+	for _, remote := range remotes {
+		pluginList[remote.Name()] = true
```

Comment:
A swarmkit level IT would have been nice, since the bug exposed here happens only with swarm mode is enabled and that too in a specific order. But given that swarm-mode IT is not available now, what you have is okay.